app/components/avo/fields/password_field/edit_component.html.erb

```diff
@@ -1,12 +1,39 @@
 <%= field_wrapper **field_wrapper_args do %>
-  <%= @form.password_field @field.id,
-    value: @field.value,
-    class: classes("w-full"),
-    data: @field.get_html(:data, view: view, element: :input),
-    disabled: disabled?,
-    placeholder: @field.placeholder,
-    style: @field.get_html(:style, view: view, element: :input),
-    autocomplete: @field.autocomplete,
-    autofocus: @autofocus
-    %>
+  <% if @field.reveal %>
+    <div class="relative" data-controller="password-visibility">
+      <%= @form.password_field @field.id,
+          value: @field.value,
+          class: classes("w-full"),
+          data: @field.get_html(:data, view: view, element: :input).merge(password_visibility_target: "input"),
+          disabled: disabled?,
+          placeholder: @field.placeholder,
+          style: @field.get_html(:style, view: view, element: :input),
+          autocomplete: @field.autocomplete,
+          autofocus: @autofocus
+      %>
```

Comment:
This looks the same as the `@form.password_field`... from the else block with the exception that `password_visibility_target: "input"` is merged to the data.

I'm wondering if we could use only 1 `password_field` helper and override the password `get_html` method to return super if `reveal` is false or `super.merge(password_visibility_target: "input")` if `reveal` is true.

Let's use the `<% if @field.reveal %>` only to render the svgs.

Reply:
we need to wrap it in a div with class `relative`; I'll test if it looks good on both and refactor accordingly